.github/workflows/build_test.yml

```diff
@@ -30,77 +33,97 @@ jobs:
+      - name: Checkout Cyclus
+        uses: actions/checkout@v4
 
-      - name: Cyclus Unit Tests
-        run: |
-          cyclus_unit_tests
+      - name: Build and Test Cyclus
+        id: build-cyclus
+        uses: docker/build-push-action@v5
+        with:
+          file: docker/Dockerfile
+          cache-from: type=registry,ref=ghcr.io/cyclus/cyclus_${{ matrix.ubuntu_versions }}_${{ matrix.pkg_mgr }}/cyclus:ci-layer-cache
+          cache-to: type=registry,ref=ghcr.io/cyclus/cyclus_${{ matrix.ubuntu_versions }}_${{ matrix.pkg_mgr }}/cyclus:ci-layer-cache,mode=max
+          tags: ghcr.io/cyclus/cyclus_${{ matrix.ubuntu_versions }}_${{ matrix.pkg_mgr }}/cyclus:ci-image-cache
+          push: true
+          build-args: |
+            pkg_mgr=${{ matrix.pkg_mgr }}
+            ubuntu_version=${{ matrix.ubuntu_versions }}
 
-      - name: Cyclus Python Tests
-        run: |
-          cd tests && python -m pytest --ignore test_main.py
+      - name: Checkout Cycamore
+        if: ${{ github.event_name == 'pull_request' && steps.build-cyclus.outcome == 'success' }}
+        uses: actions/checkout@v4
+        with:
+          repository: cyclus/cycamore
+          ref: tag-build-arg
+          path: $HOME/cycamore
 
-      - name: Unit Test Coverage Report
-        run: |
-          lcov -c --rc lcov_branch_coverage=1 -d ${PWD}/build/src -o ${HOME}/test_coverage.info --gcov-tool ${{ env.GCOV }}
-          lcov --add-tracefile ${HOME}/initial_coverage.info --add-tracefile  ${HOME}/test_coverage.info --rc lcov_branch_coverage=1 -o ${HOME}/total_coverage.info
-          lcov --remove ${HOME}/total_coverage.info -o ${HOME}/src_coverage.info --rc lcov_branch_coverage=1 "/usr/**" "/opt/**"
-          mkdir -p ${HOME}/cyclus_coverage/html
-          genhtml ${HOME}/src_coverage.info --output-directory ${HOME}/cyclus_coverage/html --branch-coverage
-          echo "artifactPath=${HOME}/cyclus_coverage/html/" >> $GITHUB_ENV
-     
-      - name: Upload Coverage HTML Artifacts
-        uses: actions/upload-pages-artifact@v2
+      - name: Build and Test Cycamore
+        id: build-cycamore
+        if: ${{ github.event_name == 'pull_request' && steps.build-cyclus.outcome == 'success' }}
+        continue-on-error: true
+        uses: docker/build-push-action@v5
         with:
-          name: cyclus-coverage-report-${{ matrix.ubuntu_versions }}-${{ matrix.pkg_mgr }}
-          path: ${{ env.artifactPath }}
+          context: $HOME/cycamore
+          file: $HOME/cycamore/docker/Dockerfile
+          cache-from: type=registry,ref=ghcr.io/cyclus/cycamore_${{ matrix.ubuntu_versions }}_${{ matrix.pkg_mgr }}/cycamore:ci-layer-cache
+          cache-to: type=registry,ref=ghcr.io/cyclus/cycamore_${{ matrix.ubuntu_versions }}_${{ matrix.pkg_mgr }}/cycamore:ci-layer-cache,mode=max
+          tags: ghcr.io/cyclus/cycamore_${{ matrix.ubuntu_versions }}_${{ matrix.pkg_mgr }}/cycamore:ci-image-cache
```

Comment:
It's not clear to me what's in the cache at this point?  Is the most recent successful cyclus build still in the cache?